db/db.go

```diff
@@ -0,0 +1,130 @@
+package db
+
+import (
+	"context"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+)
+
+// DB database
+type DB struct {
+	pool *pgxpool.Pool
+	psql sq.StatementBuilderType
+}
+
+// New creates a new DB connection
+func New(pool *pgxpool.Pool, psql sq.StatementBuilderType) *DB {
+	return &DB{
+		pool: pool,
+		psql: psql,
+	}
+}
+
+// Post ...
+type Post struct {
+	ID    int    `json:"id"`
+	Title string `json:"title"`
+	Body  string `json:"body"`
+}
+
+// GetPosts fetches all posts
+func (db *DB) GetPosts() ([]Post, error) {
+	posts := []Post{}
+
+	sql, _, err := db.psql.Select("*").From("posts").ToSql()
+	if err != nil {
+		return []Post{}, err
```

Comment:
You can return `nil` instead of `[]Post{}` everywhere, and it is equivalent here, and it's the usual practice.
```suggestion
		return nil, err
```